modules/standard/Memory.chpl

```diff
@@ -17,10 +17,96 @@
+    deallocations to be internally tracked, in order to support
+    logging them on the fly and/or producing any of several kinds of
+    memory-related reports, described below.
+
+  ``memLeaksTable``: :type:`bool`:
+    Enable memory tracking and produce detailed information about
+    leaked memory by invoking :proc:`printMemTable` implicitly when
+    when the program terminates normally.
+
+  ``memLeaks``: :type:`bool`:
+    Enable memory tracking and produce summary information about
+    leaked memory by invoking :proc:`printMemLeaksTable` implicitly
+    when the program terminates normally.
+
+  ``memLeaksLog``: :type:`c_string`:
+    Enable memory tracking and produce a report to the named file when
+    the program terminates normally.
+    The report contains the compilation command used to build the
+    program, the execution command used to run it, the summary
+    allocation statistics produced by :proc:`printMemStat`, and the
+    summary information about leaked memory produced by
+    :proc:`printMemLeaksTable`.
+
+  ``memStats``: :type:`bool`:
+    Enable memory tracking and produce summary memory statistics by
+    invoking :proc:`printMemStat` implicitly at normal program
+    termination.
+
+  ``memMax``: :type:`uint`:
+    If the value is greater than 0 (zero), enable memory tracking
+    and use the value as a limit on the number of bytes of memory
+    that can be allocated per top-level (network-connected) locale.
+    If during execution the amount of allocated memory exceeds this
+    limit on any locale, halt the program with a message saying so.
+
+  The following two config variables do not enable memory tracking;
+  they only modify how it is done.
+
+
+  ``memThreshold``: :type:`uint`:
+    If this is set to a value greater than 0 (zero), only allocation
+    requests larger than this are tracked and/or reported.
+
+  ``memLog``: :type:`c_string`:
+    Memory reporting is written to this file.  By default it is the
+    ``stdout`` associated with the process (not the Chapel channel
+    with the same name).  Setting this config variable to a file path
+    causes the reporting to be written to that named file instead.
+ */
+module Memory {
+
+pragma "no doc"
 pragma "insert line file info" extern proc chpl_memoryUsed(): uint(64);
 
+/*
+  The amount of memory returned by :proc:`locale.physicalMemory` can
+  be expressed either as individual bytes or as chunks of 2**10,
+  2**20, or 2**30 bytes.
+ */
 enum MemUnits {Bytes, KB, MB, GB};
 
+/*
+  How much physical memory is present on this locale?
+
+  Note that this is a physical quantity and does not take into account
+  things like virtual memory support which might allow for allocating
+  a larger amount, or system or user limits which might prevent
+  allocating so much.
+
+  *Note:* This is the only procedure in the :mod:`Memory` module which
+  does not need memory tracking enabled in order to work properly.
```

Comment:
"to work properly" raises the question of what happens if memory tracking is not enabled.  I suggest rewording more vaguely, like: "... which does not rely on memory tracking being enabled."